python/lsst/obs/base/fitsRawFormatterBase.py

```diff
@@ -197,7 +265,11 @@ def readInfoComponent(self, fileDescriptor, component):
         elif component == "visitInfo":
             return self.makeVisitInfo(self.readMetadata(fileDescriptor))
         elif component == "wcs":
-            return self.makeWcs(self.readMetadata(fileDescriptor))
+            metadata = self.readMetadata(fileDescriptor)
+            obsInfo = ObservationInfo(self.readMetadata(fileDescriptor))
+            detector = self.getDetector(obsInfo.detector_num)
```

Comment:
Do you think it's worthwhile to support `getDetector` returning `None`, in which case we would attempt to install a WCS from metadata, without warning about not being able to make one from cameraGeom+boresight?

If we had that, we might want to use it for DECam at the moment, because its already-distorted initial WCS may actually be better than what we can make from our cameraGeom.  The big question may be whether the new fitter @morriscb is working would be able to start from (I think it might, given that it only cares about having IWC in the starting WCS - though it may also depend on that having a certain definition that TPV's IWC might not use).

Reply:
I'll think about your first point.

I don't understand how TPV works, so I don't know if that would work. We might be better off just putting in jointcal-measured distortion terms into cameras that don't have them, anyway.

Reply:
Jim has a good point; I would like to know if the WCS we make for DECam calexps differ substantially in our current implementation vs. this new scheme. Since you seem to always make a WCS from the FITS metadata as a first step anyway, would it be possible to log a warning it differs by more than some threshold from the camera geometry computed WCS?

Reply:
We don't really have a way to quantify "how different are two SkyWcs objects" without comparing pixel->sky transforms on a grid of points, and we have no way to know which one is "better" a priori (without cross matching with a refcat). It might be useful to test whether the on-sky centers are different, but I don't know for sure whether the current approach results in CRVAL being set to the telescope boresight, or the chip center. I think it's the former for HSC, but that could be different for other cameras, which makes such comparison difficult.

It is going to take some testing to see what effect these changes have in the end. I'm hoping that if we need a better DECam distortion model, we can just compute it with jointcal.

Reply:
Since you have two SkyWcs objects just sitting there, how about something simple like "log how far apart in arcseconds/radians/whatever the on-sky center of each WCS is"? And maybe make it a warning if it's over ... I don't know ... a few arcseconds.

Reply:
If that's not feasible, because apparently CRVAL is the actual worst, I will accept an ap_verify run on the CI dataset with this ticket setup demonstrating the metrics are not wildly different from a recent CI run. I'm happy to help you with running ap_verify if you haven't done it before.